src/ddmd/statementsem.d

```diff
@@ -2490,6 +2489,24 @@ else
                         cs.error("case variables not allowed in final switch statements");
                         errors = true;
                     }
+
+                    /* Also check if the VarExp is declared in a scope outside of this one
+                     * 'scx' is set to the scope of the switch statement.
+                     */
+                    for (Scope* scx = sc; scx; scx = scx.enclosing)
+                    {
+                        if (scx.enclosing && scx.enclosing.sw == sw)
+                            continue;
+                        assert(scx.sw == sw);
```

Comment:
It looks like the above 3 lines could be replaced with simply:
```
if (scx.sw != sw) continue;
```

Reply:
That is not the intent, and that won't work.

When entering the for loop, it is already guaranteed that `scx.sw == sw`. What we want however is the first scope that set `sw`. This means traversing all enclosing scopes until `scx.enclosing.sw != sw`.

Reply:
Ah, I see. So, can you change the comment on the loop to reflect this? Your new explanation is much, much better than the code comment!

Reply:
@ibuclaw waiting for your fix to the comment!

Reply:
eh, fixed it myself

Reply:
Oops, I was going to get back to that this morning, now that I have a fully working compiler for gdc/ddmd.  Thanks!